PostHog/Utils/UIApplication+.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  UIApplication+.swift
+//  PostHog
+//
+//  Created by Yiannis Josephides on 11/11/2024.
+//
+
+#if os(iOS) || os(tvOS)
+    import UIKit
+
+    extension UIApplication {
+        static func getCurrentWindow(filterForegrounded: Bool = true) -> UIWindow? {
+            let windowScenes = UIApplication.shared
+                .connectedScenes
+                .compactMap { $0 as? UIWindowScene }
+                .filter {
+                    !filterForegrounded || $0.activationState == .foregroundActive
+                }
+
+            if #available(iOS 15.0, tvOS 15.0, *) {
+                // attempt to retrieve directly from UIWindowScene
+                if let keyWindow = windowScenes.compactMap(\.keyWindow).first {
+                    return keyWindow
+                }
+            }
+
+            // fall bar to UIWindowSceneDelegate
+            for scene in UIApplication.shared.connectedScenes where scene.activationState == .foregroundActive {
+                let windowScene = scene as? UIWindowScene
```

Comment:
This part repeats with L13-15, I guess we can assign that to a var and reuse it so we don't have to traverse the list and do the same checks again.
I also think this should be the last fallback, since the L37 is the same check as L22 but using an older API since `keyWindow` is only available >= iOS 15